### PR TITLE
feat: new a11yAudit function

### PR DESCRIPTION
**Pull request by JimLiu**
```js
browser.a11yAudit()
browser.a11yAudit({ ignore: ['document-title'], flags, config })
```

returns
```js
[
  {
    "auditId": "document-title",
    "description": "Ensures each HTML document contains a non-empty <title> element",
    "snippet": "<html>",
    "impact": "serious",
    "help": "Documents must have <title> element to aid in navigation",
    "helpUrl": "https://dequeuniversity.com/rules/axe/3.1/document-title?application=axeAPI",
    "tags": ["cat.text-alternatives", "wcag2a", "wcag242"],
    "selector": "html",
    "path": "1,HTML"
  },
  {
    "auditId": "html-has-lang",
    "description": "Ensures every HTML document has a lang attribute",
    "snippet": "<html>",
    "impact": "serious",
    "help": "<html> element must have a lang attribute",
    "helpUrl": "https://dequeuniversity.com/rules/axe/3.1/html-has-lang?application=axeAPI",
    "tags": ["cat.language", "wcag2a", "wcag311"],
    "selector": "html",
    "path": "1,HTML"
  }
]
```

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          "--timeout",
+          "999999",
+          "--colors",
+          "${file}"
+      ],
+      "console": "integratedTerminal",
+      "internalConsoleOptions": "neverOpen"
+    }
+  ]
+}
```
Comment (dbushong) [inline .vscode/launch.json]: don't think we need this checked in

Comment (JimLiu) [reply .vscode/launch.json]: Removed it.

File: lib/browser/lighthouse.js [hunk 2]
```diff
@@ -174,3 +174,42 @@ exports.assertLighthouseScore = function assertLighthouseScore(
...
+            },
+            {
+              selector: Array.isArray(node.target) ? node.target.join(' ') : '',
+              path: node.path,
+              snippet: node.snippet,
+            }
+          )
+        )
+      );
+    }, []);
```
Comment (dbushong) [inline lib/browser/lighthouse.js]: FFR, you can also fake a flatMap with something like:
```js
  return [].concat(
    ...violations.map(violation => {
      const { nodes /*, ... */ } = violation;
      return nodes.map(node => Object.assign(/* ... */));
    })
  );
```

File: test/integration/lighthouse.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const browserName = getConfig().get('browser');
+
+describe('navigation', () => {
+  if (browserName !== 'chrome') {
+    xit('Skipping lighthouse tests. They only work for chrome.');
+    return;
+  }
+
+  if (!Semver.satisfies(process.version, '>=8.0.0')) {
+    xit('Skipping lighthouse tests. They only work for node >=8.0.0.');
```
Comment (dbushong) [inline test/integration/lighthouse.test.js]: why?

Comment (dbushong) [reply test/integration/lighthouse.test.js]: and, if true, might as well write the rest of the test file using async/await
we can bump the lint requirement for this project to `groupon/node8` now that 6.x is officially unsupported

File: lib/browser/lighthouse.js (hunk 2 above)
Comment (JimLiu) [reply lib/browser/lighthouse.js]: Nice, I thought we couldn't use `...`

File: test/integration/lighthouse.test.js (hunk 3 above)
Comment (JimLiu) [reply test/integration/lighthouse.test.js]: Removed it since we only support node8+

**Comment by JimLiu** [comment]
Nice, I'll update the document tomorrow.
It would be great if it can be merged today :)

**Pull request merged**